server/src/main/java/org/opensearch/wlm/WorkloadManagementTransportInterceptor.java

```diff
@@ -8,21 +8,26 @@
 
 package org.opensearch.wlm;
 
+import org.opensearch.core.concurrency.OpenSearchRejectedExecutionException;
 import org.opensearch.tasks.Task;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.TransportChannel;
 import org.opensearch.transport.TransportInterceptor;
 import org.opensearch.transport.TransportRequest;
 import org.opensearch.transport.TransportRequestHandler;
 
+import java.util.Optional;
+
 /**
  * This class is used to intercept search traffic requests and populate the queryGroupId header in task headers
  */
 public class WorkloadManagementTransportInterceptor implements TransportInterceptor {
     private final ThreadPool threadPool;
+    private final QueryGroupService queryGroupService;
```

Comment:
Depending on the methods within `QueryGroupService`, this should be renamed to something more specific

Reply:
Can you elaborate little more on this ?